AdvancedDLSupport.Tests/Tests/Integration/CustomLoadingLogicTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿//
+//  CustomLoadingLogicTests.cs
+//
+//  Copyright (c) 2018 Firwood Software
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using AdvancedDLSupport.Loaders;
+using AdvancedDLSupport.Tests.Data;
+using AdvancedDLSupport.Tests.TestBases;
+using JetBrains.Annotations;
+using Xunit;
+
+#pragma warning disable SA1600, CS1591
+
+namespace AdvancedDLSupport.Tests.Integration
+{
+    public class CustomLoadingLogicTests : LibraryTestBase<IStringLibrary>
+    {
+        private const string LibraryName = "StringTests";
+
+        private class LibraryLoadingOverride : ILibraryLoader

```

Comment:
Don't nest these classes inside the test class. Place them in their own files in the Data directory instead - you can create a new directory there called "Implementations" or something similar.